test/WebSites/MvcTagHelpersWebSite/Components/ProductsViewComponent.cs

```diff
@@ -16,7 +16,11 @@ public IViewComponentResult Invoke(string category)
         {
             IExpirationTrigger trigger;
             var products = ProductsService.GetProducts(category, out trigger);
-            EntryLinkHelpers.ContextLink.AddExpirationTriggers(new[] { trigger });
+
+            if (EntryLinkHelpers.ContextLink != null)
```

Comment:
@Tratcher : Need to discuss with you about `EntryLinkHelpers`...the api review required this to be `internal`, but probably we should keep it `public` as I need to get access to it for this scenario...